compiler/modules/types/core.bal

```diff
@@ -171,6 +171,12 @@ public type BddMemo record {|
 
 type BddMemoTable table<BddMemo> key(bdd);
 
+type ComparableMemo record {|
+    readonly [SemType, SemType] semTypes;
```

Comment:
I think it's simpler to use multiple keys here:

```
  readonly SemType semType1;
  readonly SemType semType2;
```

then `key(semType1, semType2)`.